src/context/WidgetSettingContext.tsx

```diff
@@ -0,0 +1,209 @@
+    ) {
+      const response = await getWidgetSetting({
+        host: apiHost,
+        appId,
+        botId,
+        createUserAndChannel: false,
+      });
+
+      setBotStyle(response.botStyle);
+      setWidgetSession({
+        strategy: sessionStrategy,
+        userId: cachedSession.userId,
+        expireAt: cachedSession.expireAt,
+        channelUrl: cachedSession.channelUrl,
+        sessionToken: cachedSession.sessionToken,
+      });
+    } else {
+      const response = await getWidgetSetting({
+        host: apiHost,
+        appId,
+        botId,
+        createUserAndChannel: strategy === 'auto',
+      });
+
+      setBotStyle(response.botStyle);
+
+      if (sessionStrategy === 'auto' && response.user && response.channel) {
+        const session = {
+          strategy: sessionStrategy,
+          expireAt: response.user.expireAt,
+          userId: response.user.userId,
+          sessionToken: response.user.sessionToken,
+          channelUrl: response.channel.channelUrl,
+        };
+        setWidgetSession(session);
+        saveWidgetSessionCache({ appId, botId, data: session });
+      }
+
+      /**
+       * NOTE: We don't fully initialize the manual strategy session here.
+       * After the uikit is initializing, we should call the `initManualSession` function.
+       * */
+      if (sessionStrategy === 'manual' && injectedUserId) {
+        setWidgetSession({
+          strategy: sessionStrategy,
+          userId: injectedUserId,
+          sessionToken: undefined,
+          channelUrl: undefined,
+          expireAt: 0,
+        });
+      }
+    }
+
+    setInitialized(true);
+  }
+
+  async function initManualSession(
+    sdk: SendbirdChatWith<[GroupChannelModule]>
+  ) {
+    if (sessionStrategy === 'manual' && injectedUserId) {
+      const data = firstMessageData
+        ? JSON.stringify({ first_message_data: firstMessageData })
+        : undefined;
+
+      const channel = await sdk.groupChannel.createChannel({
+        name: createGroupChannelParams?.name,
+        coverUrl: createGroupChannelParams?.coverUrl,
+        invitedUserIds: [injectedUserId, botId],
+        isDistinct: false,
+        data,
+      });
+
+      const session = {
+        strategy: sessionStrategy,
+        expireAt: getDateNDaysLater(30),
+        userId: injectedUserId,
+        sessionToken: undefined,
+        channelUrl: channel.url,
+      };
+      setWidgetSession(session);
+      saveWidgetSessionCache({ appId, botId, data: session });
+    }
+  }
+
+  useLayoutEffect(() => {
+    // Trick to prevent duplicated request in strict mode.
+    if (inProgress.current) return;
+    inProgress.current = true;
+    initSessionByStrategy(sessionStrategy).then(() => {
+      inProgress.current = false;
+    });
+  }, [sessionStrategy]);
+
+  return (
+    <WidgetSettingContext.Provider
+      value={{ initialized, botStyle, widgetSession, initManualSession }}
+    >
+      {widgetSession ? children : null}
```

Comment:
Children will not be rendered until the session initialization is completed.
This allows you to freely call the `useWidgetSession` hook in the lower levels and use it as non-nullable.

Reply:
neat!